test-case/verify-pcm-list.sh

```diff
@@ -69,6 +71,34 @@ func_error_process()
     exit 1
 }
 
+func_compare_asound_pipeline_by_idx()
+{
+    local idx
+    local id=$(func_pipeline_parse_value $1 id)
+    local type=$(func_pipeline_parse_value $1 type)
+    local pcm=$(func_pipeline_parse_value $1 pcm)
+    # filter to catch the same id & type
+    # vierfy the pcm list for keyword: type, id, pcm
+    for idx in $(seq 0 $(expr $ASOUND_PCM_COUNT - 1))
+    do
+        if [ "$id" == "$(func_asound_pcm_parse_value $idx id)" ]; then
+            # same pcm id with different pcm name it mast mismatch
+            if [ "$pcm" != "$(func_asound_pcm_parse_value $idx pcm)" ];then
+                dlogi "TPLG: '$id: $pcm' mismatch with pcm: $(func_asound_pcm_parse_value $idx pcm)"
```

Comment:
could we also add card id here?
To make it looks more user friendly.
```
2020-02-25 10:03:21 UTC [INFO] TPLG: '0: Port0: playback' matched with pcm
```
=>
`2020-02-25 10:03:21 UTC [INFO] TPLG: 'hw:0,0: Port0: playback' matched with CARD0 PCM0P`

Reply:
`CARD0 PDM0P` how about this format: `Card 0: id=2;pcm=HDMI1;type=playback`